src/git.ts

```diff
@@ -30,18 +30,31 @@ export interface CommitListItem {
 }
 
 export function listCommits(from: string, to: string = ""): CommitListItem[] {
-  // Prints "<short-hash>;<ref-name>;<summary>;<date>"
+  // Prints "hash<short-hash> ref<ref-name> message<summary> date<date>"
   // This format is used in `getCommitInfos` for easily analize the commit.
   return execa
-    .sync("git", ["log", "--oneline", "--pretty=%h;%D;%s;%cd", "--date=short", `${from}..${to}`])
+    .sync("git", [
+      "log",
+      "--oneline",
+      "--pretty=hash<%h> ref<%D> message<%s> date<%cd>",
+      "--date=short",
+      `${from}..${to}`,
+    ])
     .stdout.split("\n")
     .filter(Boolean)
     .map((commit: string) => {
-      const parts = commit.split(";");
-      const sha = parts[0];
-      const refName = parts[1];
-      const summary = parts[2];
-      const date = parts[3];
+      const parts = commit.match(/hash<(.+)> ref<(.*)> message<(.*)> date<(.*)>/) || [];
```

Comment:
can you extract this parsing part into a dedicated function and write a few tests for it? just to make sure that we don't regress on this 🙏